lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -742,6 +737,29 @@ class DxilFinalizeModule : public ModulePass {
       // Remove store undef output.
       RemoveStoreUndefOutput(M, hlslOP);
 
+      if (!IsLib) {
+        // Set used masks for signature elements
+        MarkUsedSignatureElements(DM.GetEntryFunction(), DM);
+        if (DM.GetShaderModel()->IsHS())
+          MarkUsedSignatureElements(DM.GetPatchConstantFunction(), DM);
+      }
+
+      // Adding warning for pixel shader with unassigned target
+      if (DM.GetShaderModel()->IsPS()) {
+        DxilSignature &sig = DM.GetOutputSignature();
+        for (auto &Elt : sig.GetElements()) {
+          if (Elt->GetKind() == Semantic::Kind::Target) {
+            if (Elt->GetUsageMask() != Elt->GetColsAsMask()) {
```

Comment:
you might consider combining these conditionals with && to reduce the indenting, which for me hinders readability